wweval/R/score_real_time_outputs.R

```diff
@@ -0,0 +1,289 @@
+#'
+#' @param score_type A string indicating which score to generate, either
+#' "crps" or "wis". Note, if using crps, will score draws from nowcast and
+#' forecast. If using wis, will score only the forecasts.
+#' @param real_time_output_dir A string indicating the upper level directory
+#' where the real-time outputs live locally
+#' @param table_of_run_ids A tibble containing the forecast date, run id,
+#' and date run for each of the production runs
+#' @param locations A vector of character strings indicating the locations to
+#' pull, this should be all jurisdictions
+#' @param dates A vector of forecast dates to pull
+#' @param eval_data a tibble of hospital admissions evaluation data to be used
+#' for scoring.
+#' @param hosp_only boolean indicating if we should only pull the hospital
+#' admissions model
+#'
+#' @return A large tibble containing crps scores for every location and
+#' forecast date, conditioned on the presence of wastewater and model
+#' convergence
+#' @export
+score_real_time_outputs <- function(score_type,
+                                    real_time_output_dir,
+                                    table_of_run_ids,
+                                    locations,
+                                    dates,
+                                    eval_data,
+                                    hosp_only = FALSE) {
+  if (isTRUE(hosp_only)) {
+    model_types <- c("hosp")
+  } else {
+    model_types <- c("ww", "hosp")
+  }
+  all_scores <- c()
+
+  data_type <- ifelse(score_type == "crps", "draws", "quantiles")
+  col_name <- ifelse(score_type == "crps", "draw", "quantile")
+  for (i in seq_along(dates)) {
+    date_to_pull <- dates[i]
+    metadata <- table_of_run_ids |> dplyr::filter(
+      forecast_date == date_to_pull
+    )
+    run_id <- metadata$ids
+    date_run <- metadata$dates_run
+    for (j in seq_along(locations)) {
+      for (m in seq_along(model_types)) {
+        model_long <- ifelse(model_types[m] == "ww",
+          "site-level infection dynamics",
+          "hospital admissions only"
+        )
+        if (date_to_pull %in% c("2024-02-05", "2024-02-12")) {
+          # Assume the old  file structure
+          fp <- file.path(
+            real_time_output_dir,
+            glue::glue("output_{date_to_pull}"),
+            "raw",
+            locations[j],
+            model_long,
+            data_type,
+            date_to_pull,
+            glue::glue("run-on-{date_run}-{run_id}-{data_type}.parquet")
+          )
+          if (file.exists(fp)) {
+            # The diagnostics are not flags here, just values.
+            any_flags <- FALSE
+
+            these_preds <- arrow::read_parquet(fp) |>
+              dplyr::filter(
+                name == "pred_hosp",
+                period != "calibration"
+              ) |>
+              dplyr::select(
+                forecast_date,
+                date,
+                location,
+                value,
+                !!sym(col_name)
```

Comment:
Don't need the `sym`
```suggestion
                !!col_name
```